### PR TITLE
Protocol update

### PR DESCRIPTION
**Pull request by jaxxzer**

### CODE REVIEW

File: src/protocol/templates/ping_protocol.json [hunk 1]
```diff
@@ -1,212 +1,511 @@
...
-              ]
+  "requirements": {
+    "discovery": "discovery of connected devices, types, and firmwares",
+    "packed fields": "memory alignment"
+  },
+  "format": [
+    {
+      "name": "start1",
+      "type": "u8",
+      "index": "0"
```
Comment (patrickelectric) [inline src/protocol/templates/ping_protocol.json]: Why are you adding index ?

**Comment by patrickelectric** [comment]
I'm receiving this Critical message:
default[Critical]: ping.cpp(156) UNHANDLED MESSAGE ID: 3

**Comment by jaxxzer** [comment]
Updated/handled

File: src/protocol/templates/ping_protocol.json [hunk 2]
```diff
@@ -1,212 +1,511 @@
...
+      "index": "6"
+    },
+    {
+      "name": "dst_device_id",
+      "type": "u8",
+      "index": "7"
+    },
+    {
+      "name": "payload",
+      "type": "-",
```
Comment (patrickelectric) [inline src/protocol/templates/ping_protocol.json]: u8, because it will be a byte array

File: src/protocol/templates/ping_protocol.json [hunk 3]
```diff
@@ -1,212 +1,511 @@
...
+      "index": "8-n"
+    },
+    {
+      "name": "checksum",
+      "type": "u16",
+      "index": "(n+1)-(n+2))"
```
Comment (patrickelectric) [inline src/protocol/templates/ping_protocol.json]: "(n+1)-(n+2)"

File: src/protocol/templates/ping_protocol.json [hunk 4]
```diff
@@ -1,212 +1,511 @@
...
+          {
+            "name": "fw_version_minor",
+            "type": "u16"
+          }
+        ]
+      },
+      "device_id": {
+        "id": "1201",
+        "payload": [],
+        "deprecated": "true"
```
Comment (patrickelectric) [inline src/protocol/templates/ping_protocol.json]: We are doing this before a release, so we don't have deprecated.

File: src/protocol/templates/ping_protocol.json [hunk 5]
```diff
@@ -1,212 +1,511 @@
...
+            "name": "length_mm",
+            "type": "u32"
+          }
+        ]
+      },
+      "mode": {
+        "id": "1205",
+        "payload": [
+          {
+            "name": "auto_manual",
```
Comment (patrickelectric) [inline src/protocol/templates/ping_protocol.json]: is this working ? does not appears to be here.

File: src/protocol/templates/ping_protocol.json [hunk 6]
```diff
@@ -1,212 +1,511 @@
...
+            "name": "length_mm",
+            "type": "u32"
+          },
+          {
+            "name": "gain_index",
+            "type": "u32"
+          }
+        ]
+      },
+      "processor_degC": {
```
Comment (patrickelectric) [inline src/protocol/templates/ping_protocol.json]: processor_temp ? what deg stands for ?

File: src/protocol/templates/ping_protocol.json [hunk 7]
```diff
@@ -1,212 +1,511 @@
...
+        "id": "1213",
+        "payload": [
+          {
+            "name": "temp",
+            "type": "u16",
+            "units": "centiC",
```
Comment (patrickelectric) [inline src/protocol/templates/ping_protocol.json]: centi is value/100, value*100 is hecto or just h.
I think that the unit should be hC, stands for hecto Celsius

File: src/protocol/templates/ping_protocol.json [hunk 8]
```diff
@@ -1,212 +1,511 @@
...
+          }
+        ]
+      },
+      "pcb_degC": {
+        "id": "1214",
+        "payload": [
+          {
+            "name": "temp",
+            "type": "u16",
+            "units": "centiC",
```
Comment (patrickelectric) [inline src/protocol/templates/ping_protocol.json]: same thing here.

**Comment by jaxxzer** [comment]
ok on 1 and 2. On 3, I think no, I won't use the spreadsheet anymore, just a json editor.

File: src/protocol/templates/ping_protocol.json (hunk 4 above)
Comment (jaxxzer) [reply src/protocol/templates/ping_protocol.json]: I am leaving the note here as a reminder that the message is useless.

File: src/protocol/templates/ping_protocol.json (hunk 5 above)
Comment (jaxxzer) [reply src/protocol/templates/ping_protocol.json]: I think this was never working right. We will have to do more testing and open a new issue.

File: src/protocol/templates/ping_protocol.json (hunk 7 above)
Comment (jaxxzer) [reply src/protocol/templates/ping_protocol.json]: the value transmitted is centiC, check again.

File: src/protocol/templates/ping_protocol.json (hunk 6 above)
Comment (jaxxzer) [reply src/protocol/templates/ping_protocol.json]: degrees C, I will change it

File: src/protocol/templates/ping_protocol.json (hunk 2 above)
Comment (jaxxzer) [reply src/protocol/templates/ping_protocol.json]: The type inside of each payload is message specific. u8 is misleading imo.

**Comment by jaxxzer** [comment]
Oy, you lost my patches in ping_protocol.json according to your review!

**Comment by patrickelectric** [comment]
merging protocol-update-review.

**Pull request closed without merging**